misc/e2e-azure.sh

```diff
@@ -0,0 +1,285 @@
+export TF_VAR_worker_count=${WORKER_COUNT}
+EOF
+}
+
+stage-prepare-ssh(){
+    ssh-keygen -t rsa -N "" -f ${BUILDDIR}/${SSH_USER}.key
+
+    ssh_pub_key=$(cat ${BUILDDIR}/${SSH_USER}.key.pub)
+
+    # TODO Add after second line.
+    for n in bastion master vault; do
+      ed --quiet ${WORKDIR}/cloud-config/${n}.yaml.tmpl << EOF
+3i
+  - name: ${SSH_USER}
+    groups:
+      - "sudo"
+      - "docker"
+    ssh-authorized_keys:
+      - $(cat ${BUILDDIR}/${SSH_USER}.key.pub)
+.
+w
+q
+EOF
+    done
+
+    eval "$(ssh-agent)"
+    ssh-add ${BUILDDIR}/${SSH_USER}.key
+}
+
+stage-terraform-cloud-config() {
+  cd ${BUILDDIR}
+
+  source envs.sh
+  terraform init ../platforms/azure/giantnetes-cloud-config
+  terraform apply -state=$(mktemp) -auto-approve ../platforms/azure/giantnetes-cloud-config
+
+  cd -
+}
+
+stage-terraform-only-vault() {
+  cd ${BUILDDIR}
+
+  source envs.sh
+  terraform init ../platforms/azure/giantnetes
+  terraform apply -auto-approve -target="module.dns" ../platforms/azure/giantnetes
+  terraform apply -auto-approve -target="module.vnet" ../platforms/azure/giantnetes
+  terraform apply -auto-approve -target="module.bastion" ../platforms/azure/giantnetes
+  terraform apply -auto-approve -target="module.vault" ../platforms/azure/giantnetes
+
+  cd -
+}
+
+stage-terraform() {
+  cd ${BUILDDIR}
+
+  source envs.sh
+  terraform init ../platforms/azure/giantnetes
+  terraform apply -auto-approve ../platforms/azure/giantnetes
+
+  cd -
+}
+
+# TODO: Get rid of external dependencies and setup Vault in development mode.
+stage-vault() {
+    local base_domain=${CLUSTER}.${E2E_AZURE_LOCATION}.azure.gigantic.io
+
+    # Download Ansible playbooks for Vault bootstrap.
+    local tmp=$(mktemp -d)
+    cd $tmp
+    git clone --depth 1 --quiet https://taylorbot:${E2E_GITHUB_TOKEN}@github.com/giantswarm/hive.git
+    cd hive
+
+    # Prepare configuration for Ansible.
+    cat <<EOF > ./hosts_inventory/${CLUSTER}
+[bootstrap_node]
+vault1.${base_domain}
+
+[bootstrap_node:vars]
+ansible_python_interpreter="/root/bin/python"
+ansible_ssh_user=${SSH_USER}
+ansible_ssh_common_args='-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ProxyCommand="ssh -W %h:%p -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ${SSH_USER}@bastion1.${base_domain}"'
+EOF
+
+    cat <<EOF > ./envs/${CLUSTER}.yml
+---
+main_domain: "${base_domain}"
+allowed_domains: "kubernetes,kubernetes.default,kubernetes.default.svc,kubernetes.default.svc.local"
+bare_metal: False
+EOF
+
+    # Bootstrap insecure Vault.
+    export ANSIBLE_HOST_KEY_CHECKING=False
+    ansible-playbook -i hosts_inventory/${CLUSTER} -e dc=${CLUSTER} bootstrap1.yml
+
+    # Init Vault with one unencrypted unseal key.
+    # NOTE: sed here is to filter ANSI escape codes and color codes.
+    init_output=$(exec_on vault1 vault operator init -key-shares=1 -key-threshold=1 |\
+        sed -r -e 's/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g' -e 's/[\x01-\x1F\x7F]//g')
```

Comment:
:open_mouth:

Reply:
did not find better way unfortunately :/